src/components/Vault/VaultDepositForm.tsx

```diff
@@ -173,16 +183,22 @@ const VaultDepositForm: FC<VaultProps> = (props) => {
       logger("Approving spend", inputTokenAddress)
       approve?.write?.()
     } else {
-      logger("Depositing", amountIn)
-      depositLp?.write
-        ? depositLp.write()
-        : depositUnderlying?.write
-        ? depositUnderlying.write()
-        : tokenDepositLp?.write
-        ? tokenDepositLp.write()
-        : tokenDepositUnderlying?.write
-        ? tokenDepositUnderlying.write()
-        : null
+      if (enableDepositLp) {
```

Comment:
It's worth a few more lines of code for some clarity on what is happening when the user presses submit.